### ISSUE TITLE
How to retain state of a revisited page?

### ISSUE THREAD

**Issue by benny856694** [tryphotino/photino.Blazor#145]
I want to retain the sate of the page, is there a good way?
using a singleton class to hold the sate and inject to the page?

**Comment by MattParkerDev** [comment]
Yes, a simple AppState pattern can be used if you want to load the data for a page from in memory.

```csharp
public class AppState
{
	public string? MyProperty { get; set; }
}
```

Then yes, register it as a singleton:

```csharp
// Program.cs
appBuilder.Services.AddSingleton<AppState>();
```

Then it can be injected into a page:

```razor
@* MyPage.razor *@
@inject AppState AppState
```

And used like so:

```razor
@code {
	private string? _myProperty;

	protected override void OnInitialized()
	{
		_myProperty = AppState.MyProperty;
	}
}
```

Be aware of the gotcha that if you bind any UI elements to properties on AppState, updating any values in AppState from a different view will not cause a re-render of the current view - you will need to manually trigger it yourself somehow.

**Comment by MikeYeager** [comment]
Can this be closed then?
